adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -407,13 +411,15 @@ func tombstoneDeletionConcurrency() int {
 	return concurrency
 }
 
-func (h *hnsw) reassignNeighborsOf(deleteList helpers.AllowList, breakCleanUpTombstonedNodes breakCleanUpTombstonedNodesFunc) (ok bool, err error) {
+func (h *hnsw) reassignNeighborsOf(ctx context.Context, deleteList helpers.AllowList,
+	breakCleanUpTombstonedNodes breakCleanUpTombstonedNodesFunc,
+) (ok bool, err error) {
 	h.RLock()
 	size := len(h.nodes)
 	h.RUnlock()
 
-	g, ctx := enterrors.NewErrorGroupWithContextWrapper(h.logger, h.shutdownCtx)
-	ctx, cancel := context.WithCancel(ctx)
+	g, shutdownContext := enterrors.NewErrorGroupWithContextWrapper(h.logger, h.shutdownCtx)
```

Comment:
**question:** double check,  is that meant to be `h.shutdownCtx` or `ctx` ?

Reply:
Agree, also it doesn't look like we're using the `ctx` we're passed as parameter. Is `h.shutdownCtx` different ?

Reply:
It was previously called `ctx` and I renamed the existing `ctx` to `shutdownContext` as that what it' seems to be doing. Then renaming all previous uses. The new usage (`ctx` passed in as an arg) is used in line 450.

There is a good chance we could use the same context, yeah, but I'm not really sure how to verify this.

Reply:
Pushed a new commit that should simplify things.